File: rust-client/examples/register_private_wallet.rs
use anyhow::Result;
use rust_client_example::{cli_keypair, setup, SetupContext};
use solana_keypair::Keypair;
use solana_signer::Signer;
use solana_system_interface::instruction::transfer;
use zolana_client::{sign_transaction, Rpc, SolanaRpc, ZolanaClient};
use zolana_keypair::ShieldedKeypair;
use zolana_wallet::{build_registration_transaction_sync, is_wallet_registered_sync};

const FUND_LAMPORTS: u64 = 10_000_000;

fn main() -> Result<()> {
    let SetupContext {
        rpc_url,
        indexer_url,
        prover_url,
        ..
    } = setup()?;

    // Connect to the RPC, indexer, and prover.
    let client = ZolanaClient::from_urls(SolanaRpc::new(rpc_url), &indexer_url, prover_url)?;

    // Initialize the sender's private wallet and local authority
    // to decrypt transactions and sync balances.
    // The Solana signer and private wallet are derived from the same Ed25519 seed.
    let sender = ShieldedKeypair::from_keypair(&Keypair::new())?;

    // The SDK hands back a message; the CLI functions as sponsor to sign and send.
    let payer = cli_keypair()?;
    client.create_and_send_transaction(
        &[transfer(&payer.pubkey(), &sender.pubkey(), FUND_LAMPORTS)],
        payer.pubkey(),
        &[&payer],
        client.compute_budget(),
    )?;
    if let Some(registration) = build_registration_transaction_sync(
        &client,
        sender.pubkey(),
        &sender.shielded_address()?,
        None,
        None,
    )? {
        let registration = sign_transaction(registration, &[&sender])?;
        client.process_transaction(registration)?;
    }

    assert!(is_wallet_registered_sync(&client, sender.pubkey())?);

    println!("ok private wallet solana_address={}", sender.pubkey());
    Ok(())
}

File: typescript-client/examples/register_private_wallet.ts
import {
  ShieldedKeypair,
  SigningKey,
  buildRegistrationTransaction,
  createZolanaClient,
} from "@heliuslabs/zolana";
import { isWalletRegistered } from "@heliuslabs/zolana/wallet";

import {
  cliKeypair,
  sendAndConfirmFactory,
  sendTransactionFactory,
  setup,
  transferLamportsInstruction,
} from "../src/lib.js";

const FUND_LAMPORTS = 10_000_000n;

async function main(): Promise<void> {
  const { clientConfig } = await setup();

  // Connect to the RPC, indexer, and prover.
  const client =
    await createZolanaClient(clientConfig);

  // Initialize the sender's private wallet and local authority
  // to decrypt transactions and sync balances.
  // The Solana signer and private wallet are derived from the same Ed25519 seed.
  const sender = ShieldedKeypair.fromKeypair(
    SigningKey.generate("ed25519"),
  );
  const senderSigner = sender.toSolanaSigner();

  // The SDK hands back a transaction; the CLI functions as sponsor to sign and send.
  const payer = ShieldedKeypair.fromKeypair(
    await cliKeypair(),
  ).toSolanaSigner();
  await sendAndConfirmFactory(
    client,
    payer,
  )([
    transferLamportsInstruction(
      payer.address,
      senderSigner.address,
      FUND_LAMPORTS,
    ),
  ]);
  const registration =
    await buildRegistrationTransaction({
      client,
      owner: senderSigner.address,
      address: sender.shieldedAddress(),
    });
  if (registration !== undefined) {
    await sendTransactionFactory(
      client,
      senderSigner,
    )(registration);
  }

  const registered = await isWalletRegistered({
    rpc: client,
    owner: senderSigner.address,
  });
  if (!registered) {
    throw new Error(
      "expected the wallet to be registered",
    );
  }

  console.log(
    `ok private wallet solana_address=${senderSigner.address}`,
  );
}

await main();
